virtwho/register.py

```diff
@@ -1,5 +1,281 @@
-"""
-class SubscriptionManager()
-class RHSMAPI()
-class SatelliteCLI()
-"""
+import time
+from virtwho import logger, FailException
+from virtwho.configure import get_register_handler
+from virtwho.ssh import SSHConnect
+
+
+class SubscriptionManager:
+
+    def __init__(self, host, username, password, port=22, register_type='rhsm',
+                 org=None, activation_key=None):
+        """
+        Define the global variables.
+        :param host: ip/hostname to run subscription-manager command.
+        :param username: account username of the host
+        :param password: password to access the host
+        :param port: port to access the host
+        :param register_type: rhsm or satellite, rhsm as default
+        :param org: organization of the entitlement server
+        :param activation_key: activation_key of the entitlement server
+        """
+        self.host = host
+        self.register_type = register_type
+        self.rhsm_conf = '/etc/rhsm/rhsm.conf'
+        self.ssh = SSHConnect(host=host, user=username, pwd=password, port=port)
+        register = get_register_handler(register_type)
+        self.server = register.server
+        self.username = register.username
+        self.password = register.password
+        self.port = register.port
+        self.prefix = register.prefix
+        self.org = org or register.default_org
+        self.activation_key = activation_key or register.activation_key
+
+    def register(self):
+        """
+        Register host by subscription-manager command
+        """
+        self.unregister()
+        cmd = f'subscription-manager register ' \
+              f'--serverurl={self.server}:{self.port}{self.prefix} ' \
+              f'--org={self.org} '
+        if 'satellite' in self.register_type and self.activation_key:
+            cmd += f'--activationkey={self.activation_key} '
+        else:
+            cmd += f'--username={self.username} ' \
+                   f'--password={self.password} '
+        self.satellite_cert_install()
```

Comment:
it's more readable:
```
if f 'satellite' in self.register_type:
    self.satellite_cert_install()
```

Reply:
Done it and removed the judge in self.satellite_cert_install() internal, which is better.